Create README.md (#6)
upfluence/ubuild: README.md

# ubuild

> Ubuild is our docker image builder utilities. It packs several tools and workflow that we use through most of our projects. Ubuild objective is two fold, build the docker image and push it to docker hub.

Reviewer: `ubuild` let's keep the simple case.  

and "ubuild is upfluence's release tool. It allows you to:
* Generate a release
* Compile programs
*  Build containers
* Push those containers to a remote registry

Everything in one command.

Author: ```suggestion
ubuild is our release tool. It allows you to:

    Generate a release
    Compile programs
    Build containers
    Push those containers to a remote registry

Everything in one command.
```

## Install

`go get  github.com/upfluence/ubuild`

## Prerequisite

Before running ubuild  you must export a GITHUB_TOKEN and set RELEASE to true  into your environment and be logged in into a docker hub account with access to upfluence image registry.

## Configuration
```yaml
type: <lang> go, rb, py, frontend
verbose: <true/false>
repository: <github_path>
# needed only for compiled language
compiler:
  binaries:
    - path: <path_to_entrypp>
  dist:
  CGO: 
  args: # map of arguments to pass to the compiler
    key: val
    ... 
docker:
  dockerfile: <path_to_dockerfile> # Dockerfile by default
  image: <image_name>
  tags: # additional tags
    key: val
    ...
deployer:
  url:
  envs:
    key: val
    ...
```

For example this configuration will build an image based on a python package, update the release of upfluence/ner_analyser and update to the docker image named "ner-analyser".
```yaml
type: py
verbose: true

> repository: <org>/<repositoryW

Reviewer 2: `<repository>`

Author: ```suggestion
repository: <org>/<repository>
```

docker:
  image: "<image_name>"
```

## Circle CI

> To use ubuild with circle-ci you must use the golang primary image have run go get  in a step of you job. The sensitive environment variable should be set in the project settings.

Reviewer 2: `get in`

Author: ```suggestion
To use ubuild with circle-ci you must use the golang primary image have run go get  in a step of you job. The sensitive environment variable should be fetched from the project settings.
```

Author: @tgallice Are you ok with the suggestion ?
